### DISCUSSION TITLE
Question: How could I process some comments in my text but ignore others?

### DISCUSSION THREAD

**Discussion by RobertoRoos** [textX/textX#407] [Q&A] [answered]
I am working on a docstring parser for PLC code (Structured Text). So of course comments are relevant and I cannot simply ignore them all. However, random comments in unexpected places should be ignored as you would normally.

`textx` has the reserved `Comment` command (typically `"/\/\/.*$/"`), to ignore all comments. This seems to be done before any other rules are applied, so this would remove also the docstring comments I am looking for.

Can I lower the priority of the `Comment` command? Is there some other way to ignore all comments but a select view?

Simplified example code:

```
// Ignored
// Important
FUNCTION_BLOCK FB_MyBlock // Ignored
// Ignored
VAR_INPUT // Ignored
    someInput       : LREAL;  // Important
    // Ignored
    otherInput      : BOOL;                    // Important
END_VAR
// Ignored
```

Matching TextX grammar:

```
Declaration:
    (function=Function)?
;

Function:
    (comment=/\/\/.*$/)?
    // I hoped ^ this would take priority over `Comment`, but it seems not
    FUNCTION_BLOCK name=ID
;

Comment: /\/\/.*$/;
```

But in the parsed result `function.comment` is always `None`.

Thanks. If this should be on Stackoverflow instead, I'll move it.

**Comment by igordejanovic** [comment, marked as answer]
Comments are currently ignored but there is an open issue to enable keeping them in the model. It would first require change in the Arpeggio parser to keep comments in the parse tree (see [this line](https://github.com/textX/Arpeggio/blob/master/arpeggio/__init__.py#L743)). See also discussions #54, https://github.com/textX/Arpeggio/issues/56, #204.

**Reply by RobertoRoos** [reply]
I see, thanks for your reply.

Until then I'm working around this by just adding a lot of comment-matching groups, e.g.:

```
Declaration:
    (function=Function)?
;

Function:
    comments*=CommentLine
    FUNCTION_BLOCK name=ID
    (CommentLine)*
;

CommentLine: /\/\/.*$/;

// Comment: CommentLine; //< Disabled!
```

It is then up higher level logic to find the relevant doc comments I actually want to keep.
